staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1519,6 +1531,51 @@ func (vs *VSphere) NodeAdded(obj interface{}) {
 	if err := vs.nodeManager.RegisterNode(node); err != nil {
 		klog.Errorf("failed to add node %+v: %v", node, err)
 	}
+	vs.setNodeZoneLabels(node)
+}
+
+func (vs *VSphere) setNodeZoneLabels(node *v1.Node) {
+	nodeZone := node.ObjectMeta.Labels[v1.LabelTopologyZone]
+	nodeRegion := node.ObjectMeta.Labels[v1.LabelTopologyRegion]
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	if vs.isSecretInfoProvided && vs.isZoneEnabled() {
+		zone, err := vs.GetZoneByProviderID(ctx, node.Spec.ProviderID)
+		if err != nil {
+			klog.Warningf("Can not get Zones from vCenter: %v", err)
```

Comment:
```suggestion
			klog.InfoS("Can not get Zones from vCenter", "err", err)
```

Reply:
Not sure we should do this upgrade piecemeal. Seems like this should be done for the whole section.